rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/CovidRDTBarcodeFactory.java

```diff
@@ -41,10 +42,9 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
 
                 String[] individualVals = splitCSV(barcodeVals);
                 populateRelevantFields(individualVals);
-                moveToNextStep(convertDate(individualVals[1], "YYYY-MM-dd"));
-            } catch (JSONException e) {
-                Timber.e(e);
-            } catch (ParseException e) {
+                final int sensorTriggerIndex = 4;
+                moveToNextStep(Boolean.parseBoolean(individualVals[sensorTriggerIndex]), convertDate(individualVals[1], "YYYY-MM-dd"));
```

Comment:
`individualVals[sensorTriggerIndex]`, it is possible for this to be an empty string or null. In this case this should default to true. Please add a private method to perform this logic and invoke the method here.

Reply:
fixed.